addOns/sstiscanner/src/main/java/org/zaproxy/zap/extension/sstiscanner/SSTIScanner.java

```diff
@@ -0,0 +1,501 @@
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.core.scanner.Plugin;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.ascanrules.PersistentXSSUtils;
+import org.zaproxy.zap.model.TechSet;
+
+/**
+ * Active Plugin for Server Side Template Injection testing and verification.
+ *
+ * @author DiogoMRSilva (2018)
+ */
+public class SSTIScanner extends AbstractAppParamPlugin {
+
+    /** Prefix for internationalised messages used by this rule */
+    private static final String MESSAGE_PREFIX = "sstiscanner.sstiplugin.";
+
+    static final String DELIMITER = "zj";
+
+    private static final float SIMILARITY_THRESHOLD = 0.75f;
+
+    private static String[] error_polyglots = {"<%={{={@{#{${zj}}%>", "<th:t=\"${zj}#foreach"};
+    // Innocuous versions of the polyglots may be useful in the future if more complex error
+    // detection is required
+    // "\\{*\\<\\%\\=\\{\\{\\=\\{\\@\\{\\#\\{\\$\\{zj\\}\\}\\%\\>\\*}"
+    // the first and last tag are the commentaries from twig
+    // "\\\"\\<th\\:t\\=\\$\\{zj\\}\\#\\foreach"
+
+    private static final TemplateFormat[] template_formats = {
+        new TemplateFormat(" ", " "),
+        new TemplateFormat("{", "}"),
+        new TemplateFormat("${", "}"),
+        new TemplateFormat("#{", "}"),
+        new TemplateFormat("{#", "}"),
+        new TemplateFormat("{@", "}"),
+        new TemplateFormat("{{", "}}"),
+        new TemplateFormat("{{=", "}}"),
+        new TemplateFormat("<%%=", "%%>"),
+        new TemplateFormat("#set($x=", ")${x}"),
+        new TemplateFormat("<p th:text=\"${", "}\"></p>"),
+        new TemplateFormat(
+                "{", "}", "{@math key=\"%d\" method=\"multiply\" operand=\"%d\"/}"), // Dustjs
+        new DjangoTemplateFormat(),
+        new goTemplateFormat()
+    };
+
+    private static final String[] ways_to_fix_code_syntax = {"\"", "'", "1", ""};
+
+    // Logger instance
+    private static final Logger log = Logger.getLogger(SSTIScanner.class);
+
+    /**
+     * Get the unique identifier of this plugin
+     *
+     * @return this plugin identifier
+     */
+    @Override
+    public int getId() {
+        return 90035;
+    }
+
+    /**
+     * Get the name of this plugin
+     *
+     * @return the plugin name
+     */
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    /**
+     * Give back specific plugin dependencies (none for this)
+     *
+     * @return the list of plugins that need to be executed before
+     */
+    @Override
+    public String[] getDependency() {
+        return new String[] {};
+        // TODO activate when start to work properly
+        // return new String[] {"TestPersistentXSSSpider"};
+    }
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        return true;
```

Comment:
Was this done on purpose?
Is it temporary?
(Should have a comment?)

Reply:
It is doing nothing because is the same as the default. I am going to remove it.